### DISCUSSION TITLE
Any good examples of how to export blendshapes?

### DISCUSSION THREAD

**Discussion by AMGI-Pipeline** [Autodesk/maya-usd#1148] [Q&A] [answered]
I noticed in the release notes for version 0.7.0, Blendshape export is supported.  are there any examples of how to export animation with Blendshapes with usd?  So far, I have been unsuccessful.  Joint animation translates over, but Blendshapes are gone.

**Comment by dgovil** [comment, marked as answer]
The tests have an example.
https://github.com/Autodesk/maya-usd/tree/dev/test/lib/usd/translators/UsdExportBlendShapesTest
https://github.com/Autodesk/maya-usd/blob/dev/test/lib/usd/translators/testUsdExportBlendshapes.py

But it should work by using a command like:
`cmds.mayaUSDExport(f=temp_file, v=True, sl=True, ebs=True, skl="auto", skn="auto")`

The last 3 flags are important.
Note that this would only ensure that the blendshapes are in the resulting usd file. It doesn't mean Maya can import the blendshapes back in, nor that other apps/engines support Blendshape import at this moment.

**Reply by AMGI-Pipeline** [reply]
should the blendshape animation show up in usdview?

**Reply by AMGI-Pipeline** [reply]
Thanks so much for the quick response BTW!

**Reply by dgovil** [reply]
They should show up in usdview, though I forget if usdview still requires baking down the UsdSkel data before it can be viewed.

The best way to diagnose it would be first to export to a `.usda` file and see if the Blendshape data is represented in there.

**Reply by pmolodo** [reply]
usdview doesn't need baking anymore - it should show up properly there without any extra work.

**Reply by AMGI-Pipeline** [reply]
the unittest examples do not include joint animation combined with blendshape animation that is not baked.  I would love to see an example of that process.

**Reply by AMGI-Pipeline** [reply]
Thank You @dgovil,

After going through testUsdExportBlendshapes.py, I was able to figure out the necessary steps to export the Blendshape and joint animation with USD.  I did not need to bake anything.

**Reply by kxl-adsk** [reply]
This is great to hear “. Is there something we should clarify when documenting blendshape support?

**Reply by AMGI-Pipeline** [reply]
@kxl-adsk We're trying to export character meshes and animation with joints and blend shapes, and I haven't seen any examples of that.  It took me a day of going through the unit tests to figure out which nodes needed to be selected, which nodes did not need to be selected for export, and what the hierarchy of the selected nodes needs to look like.

One other thing that I still haven't figured out is why my polygonal mesh looks faceted, like it's lost it's smoothing groups when i look at the exported usd stage in usdview, but when I open the stage in Maya, the polygons look smooth.

**Reply by ysiewappl** [reply]
> One other thing that I still haven't figured out is why my polygonal mesh looks faceted, like it's lost it's smoothing groups when i look at the exported usd stage in usdview, but when I open the stage in Maya, the polygons look smooth.

Try adding `-defaultMeshScheme "catmullClark"` to your export command.

**Reply by AMGI-Pipeline** [reply]
Another Issue that I'm running into is that the joint heirarchy seems to get lost.  If I re-import the usd into maya, all of the joints load located at the origin.

**Comment by ysiewappl** [comment]
Please check if you got warnings/API errors in the Script Editor/stderr (i.e. launch Maya from a terminal and check the output, or if you're on Windows, check the Output Window). It might give a clue as to what might have gone wrong. If you have warnings, try using the `ignoreWarnings` flag as well and see if that gets you any farther.
